explorer/helpers.go

```diff
@@ -0,0 +1,84 @@
+package explorer
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog/log"
+	"github.com/threefoldtech/zos/pkg/gridtypes"
+	"github.com/threefoldtech/zos/pkg/rmb"
+)
+
+// NewNodeClient Creates new node client from the twin id
+func NewNodeClient(nodeTwin uint32, bus rmb.Client) *NodeClient {
+	return &NodeClient{nodeTwin, bus}
+}
+
+func getNodeTwinID(nodeID string) uint32 {
+	queryString := fmt.Sprintf(`
+	{
+		nodes(limit:10, where:{nodeId_eq:%s}){
+		  twinId
+		}
+	}
+	`, nodeID)
+
+	result := []byte(query(queryString))
+
+	var res NodeResult
+	err := json.Unmarshal(result, &res)
+
+	if err != nil {
+		log.Error().Err(errors.Wrap(err, "couldn't parse json")).Msg("connection error")
+	}
+	nodeStats := res.Data.NodeResult
+	if len(nodeStats) > 0 {
+		return nodeStats[0].TwinID
+	}
+	return 0
+
+}
+
+// NodeStatistics Returns actual node Statistics from the node itself over the msgbus
+func (n *NodeClient) NodeStatistics(ctx context.Context) (total CapacityResult, err error) {
+	const cmd = "zos.statistics.get"
+	var result struct {
+		Total gridtypes.Capacity `json:"total"`
+		Used  gridtypes.Capacity `json:"used"`
+	}
+	if err = n.bus.Call(ctx, n.nodeTwin, cmd, nil, &result); err != nil {
+		return
+	}
+
+	return result, nil
+}
+
+func query(jsonQuery string) string {
+	jsonData := map[string]string{
+		"query": jsonQuery,
+	}
+	jsonValue, _ := json.Marshal(jsonData)
+	request, err := http.NewRequest("POST", "https://explorer.devnet.grid.tf/graphql/", bytes.NewBuffer(jsonValue))
+	request.Header.Set("Content-Type", "application/json")
+	if err != nil {
+		log.Error().Err(errors.Wrap(err, "Failed to connect to graphql network")).Msg("connection error")
+	}
+
+	client := &http.Client{Timeout: time.Second * 10}
+	response, err := client.Do(request)
+	if err != nil {
+		log.Error().Err(errors.Wrap(err, "Failed to connect to graphql network")).Msg("connection error")
+	}
+	defer response.Body.Close()
+	if err != nil {
+		log.Error().Err(errors.Wrap(err, "Failed to connect to graphql network, Response Failed")).Msg("connection error")
+	}
+	data, _ := ioutil.ReadAll(response.Body)
+	return string(data)
```

Comment:
While this will still work, i don't like reading the full body then copy the entire string then later load this with json. Instead (and would be more efficient) to do it as following

```go
func query(query string, result interface{}){
  // do all exactly the same 
  // but then

  if err := json.NewDecoder(response.Body).Decode(result); err != nil {
     return err
  } 
  return nil
}
// then u can call query as 
func test(){
  var object MyConcreteType
   if err := query("my query", &object); err != nil {
      return err
   }
}
```

UPDATE: I noticed that sometimes u need to forward result as is to the caller (directly from graphql). In that case u can optimize this like

```go
func baseQuery(query) (io.ReadClose, error) {
  /// do Not defer response.Body.Close()
  return response.Body
} 

func query(q string, result interface{}) error {
 // as above but does
 response, err  := baseQuery(q)
/// process error
defer response.Close()
// json decode as above.
}

// and finally
func queryProxy(q, w io.Writer) { // or better name
  // as above but does
   response, err  := baseQuery(q)
  /// process error
   defer response.Close()
  return io.Copy(w, response.Body)
 }
```

This way u don't hold large strings in memory instead data is either copied to output stream directly, OR loaded into concrete objects